src/flb_upstream.c

```diff
@@ -796,7 +796,10 @@ int flb_upstream_conn_timeouts(struct mk_list *list)
                  * waiting for I/O will receive the notification and trigger
                  * the error to it caller.
                  */
-                shutdown(u_conn->fd, SHUT_RDWR);
+                if (u_conn->fd != -1) {
```

Comment:
Somewhat nitpick: I see the `!=` pattern mentioned in the fluent-bit style guide, but imo, it would be safer to check that a fd is non-negative with ` > -1` or `>= 0`